inference.py

```diff
@@ -225,47 +228,86 @@ def main(params: dict):
     :param params: (dict) Parameters found in the yaml config file.
 
     """
-    # SET BASIC VARIABLES AND PATHS
     since = time.time()
-    task = params['global']['task']
-    img_dir_or_csv = params['inference']['img_dir_or_csv_file']
-    chunk_size = get_key_def('chunk_size', params['inference'], 512)
-    num_classes = params['global']['num_classes']
-    num_classes_corrected = add_background_to_num_class(task, num_classes)
-    num_bands = params['global']['number_of_bands']
+
+    # MANDATORY PARAMETERS
+    img_dir_or_csv = get_key_def('img_dir_or_csv_file', params['inference'], expected_type=str)
+    if not Path(img_dir_or_csv).exists():
+        raise FileNotFoundError(f'Couldn\'t locate file or directory "{img_dir_or_csv}" '
+                                f'containing imagery for inference')
+    state_dict = get_key_def('state_dict_path', params['inference'], expected_type=str)
+    if not Path(state_dict).is_file():
+        raise FileNotFoundError(f'Couldn\'t locate state_dict of model "{state_dict}" to be used for inference')
+    task = get_key_def('task', params['global'], default='segmentation', expected_type=str)
+    model_name = get_key_def('model_name', params['global'], expected_type=str).lower()
+    num_classes = get_key_def('num_classes', params['global'], expected_type=int)
+    num_bands = get_key_def('number_of_bands', params['global'], expected_type=int)
+
+    # OPTIONAL PARAMETERS
+    chunk_size = get_key_def('chunk_size', params['inference'], default=512, expected_type=int)
+    dontcare_val = get_key_def("ignore_index", params["training"], default=-1, expected_type=int)
+    num_devices = get_key_def('num_gpus', params['global'], default=0, expected_type=int)
+
+    # SETTING OUTPUT DIRECTORY
     working_folder = Path(params['inference']['state_dict_path']).parent.joinpath(f'inference_{num_bands}bands')
-    num_devices = params['global']['num_gpus'] if params['global']['num_gpus'] else 0
     Path.mkdir(working_folder, parents=True, exist_ok=True)
-    print(f'Inferences will be saved to: {working_folder}\n\n')
 
+
+    # mlflow logging
+    mlflow_uri = get_key_def('mlflow_uri', params['global'], default=None, expected_type=str)
+    if mlflow_uri:
```

Comment:
this is a quick fix to integrate your comment, @victorlazio109 
is it ok to use the existence of mlflow_uri as a flag to activate all logging?